### PR TITLE
[release-23.05] matrix-appservice-irc: 0.38.0 -> 1.0.1

### PR DESCRIPTION
**Pull request by mweinelt**
https://github.com/matrix-org/matrix-appservice-irc/releases/tag/1.0.0
https://github.com/matrix-org/matrix-appservice-irc/releases/tag/1.0.1

(cherry picked from commit 69bc2ad72fb743b7581a01c59e73a839a6d83c87)

Backport #246188 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/matrix-synapse/matrix-appservice-irc/default.nix [hunk 1]
```diff
@@ -1,32 +1,79 @@
...
+    yarn --offline build
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir $out
+    cp app.js config.schema.yml $out
```
Comment (mweinelt) [inline pkgs/servers/matrix-synapse/matrix-appservice-irc/default.nix:65]: Needs to copy package.json, maybe yarn.lock as well. Else the following happens during startup:

> ERROR:CLI Error: Could not find package.json up from /nix/store/qvpjwskw6j9n4g8749snwj5b6nk9xsb1-matrix-appservice-irc-1.0.1/bin/matrix-appservice-irc

**Comment by Ralith** [comment]
Any progress on this? I understand it to be a critical security fix.

**Comment by mweinelt** [comment]
@ofborg test matrix-appservice-irc

**Comment by mweinelt** [comment]
>        > /nix/store/vfdg65hiv4bwls48588msw8la7452w2q-stdenv-linux/setup: line 1608: fixup-yarn-lock: command not found

cc @gdamjan @lilyinstarlight

what do we do?

**Comment by lilyinstarlight** [comment]
> what do we do?

I mean #214062 should be completely backwards compatible (it just adds a new command), so we could just backport it

**Comment by gdamjan** [comment]
> so we could just backport it

soooo… who's gonna do it?

**Comment by lilyinstarlight** [comment]
> soooo… who's gonna do it?

Backport PR #247106 has been opened

**Comment by mweinelt** [comment]
@ofborg eval

**Pull request merged**